private/buf/cmd/buf/buf.go

```diff
@@ -170,6 +170,17 @@ func NewRootCommand(name string) *appcmd.Command {
 					registrylogin.NewCommand("login", builder),
 					registrylogout.NewCommand("logout", builder),
 					registrycc.NewCommand("cc", builder, ``, false),
+					{
+						Use:   "sdk",
+						Short: "Manage Generated SDKs",
+						SubCommands: []*appcmd.Command{
+							goversion.NewCommand("go-version", builder),
+							mavenversion.NewCommand("maven-version", builder),
+							npmversion.NewCommand("npm-version", builder),
+							swiftversion.NewCommand("swift-version", builder),
+							pythonversion.NewCommand("python-version", builder),
```

Comment:
Having separate largely identical commands doesn't seem great long term. Ideally to support a new SDK, we'd just allow a new value for `registry` and not have to define a new command.

I checked all the implementations and they are all identical except for the example plugin shown (sometimes with an invalid version as mentioned in previous comment) and the swift CLI takes a `appext.SubCommandBuilder` instead of an `appext.Builder` like all the other ones.

Reply:
To answer your question, I think yes, the versioning scheme depends on the registry type and we want the flexibility for this to change across SDK implementations.

But, as a user, it's quite nice to have a single command because the shape of the response is the same between all of them. They all return a version whose schema may or may not change depending on the plugin's registry type. I agree @pkwarren's proposal to collapse these.

If we add a `--registry-type=npm,go,maven` and make it mandatory then at least it's clear which version output you'll get, although it's a bit redundant since a plugin technically has this information.

Reply:
The server knows whether a plugin is "generated SDK compatible", so for most users it should be sufficient to:

```
buf registry sdk version --plugin=protocolbuffers/go:v1.30.0 --module=bufbuild/buf:latest
```

and get back a version, or an error saying "This plugin is incompatible with generated sdks".

In the JSON payload, we could also return a "registry type" if users wanted to programmatically parse the versioning schema based on the response.

I think simplifying this makes it easier for users to grok and us to maintain?